src/games/strategy/engine/data/Card.java

```diff
@@ -0,0 +1,110 @@
+package games.strategy.engine.data;
+
+import java.io.Serializable;
+
+import games.strategy.engine.data.annotations.GameProperty;
+import games.strategy.net.GUID;
+
+public class Card extends GameDataComponent implements Serializable {
+  private static final long serialVersionUID = -2675788203388970185L;
+  private final GUID uid;
+  private PlayerID owner;
+  private String name;
+
+  /**
+   * Creates new Card. Should use a call to UnitType.create(). Owner can be null
+   */
+  protected Card(final String name, final PlayerID owner, final GameData data) {
+    super(data);
+    uid = new GUID();
+    setName(name);
+    setOwner(owner);
+  }
+
+  public GUID getID() {
+    return uid;
+  }
+
+  public PlayerID getOwner() {
+    return owner;
+  }
+
+  /**
+   * can be null.
+   */
+  @GameProperty(xmlProperty = false, gameProperty = true, adds = false)
+  void setOwner(PlayerID player) {
+    if (player == null) {
+      player = PlayerID.NULL_PLAYERID;
+    }
+    owner = player;
+  }
+
+  public final String getName() {
+    return name;
+  }
+
+  public final void setName(final String name) {
+    this.name = name;
+  }
+
+  @Override
+  public boolean equals(final Object o) {
+    if (!(o instanceof Card)) {
+      return false;
+    }
+    final Card other = (Card) o;
+    return this.uid.equals(other.uid);
+  }
+
+  @Override
+  public int hashCode() {
+    if (name != null || owner == null || uid == null || this.getData() == null) {
+      final String text =
+          "Card.toString() -> Possible java de-serialization error: Card "
+              + (name == null ? "with UNKNOWN name " : "'" + name + "'") + " owned by " + (owner == null
+                  ? "UNKNOWN OWNER" : owner.getName())
+              + " with id: " + getID();
+      CardDeserializationErrorLazyMessage.printError(text);
+      return 0;
+    }
+    return uid.hashCode();
+  }
+
+  @Override
+  public String toString() {
+    // TODO: Check if this problem from Unit class also exists here
```

Comment:
This toString() method is really complex. Pls just use StringBuilder (maybe called toStringBuilder) there is a guava class out there to help. Same for hashcode.